draft-ietf-httpbis-http2-secondary-certs.md

```diff
@@ -753,6 +753,80 @@ any stream which references the failing certificate in question or process the
+are restrictions asserted by the requester of the certificate and are not
+verified by the CA.  Conforming CAs SHOULD mark the allowedPrimaryDomain
+extension as non-critical.
+
+Each allowed primary domain is represented as an IA5String. The name MUST be in
+the "preferred name syntax", as specified by Section 3.5 of {{!RFC1034}} and as
+modified by Section 2.1 of {{!RFC1123}}.  Note that while uppercase and
+lowercase letters are allowed in domain names, no significance is attached to
+the case. In addition, while the string " " is a legal domain name,
+affiliatedDomain extensions MUST NOT contain " ".  Rules for encoding
+internationalized domain names are specified in Section 7.2 of {{!RFC5280}}.
+
+If the affiliatedDomain extension is present, the sequence MUST contain at least
+one entry.  Unlike the subject field, conforming CAs MUST NOT issue certificates
+with an affiliatedDomain extension containing empty GeneralName fields.  Clients
+that encounter such a certificate when processing a certification path MUST
+ignore that extension.
+
+The wildcard character "*" MAY be used to represent that any set of subdomains
+is acceptable.  This character MUST be the entirety of the first label if used,
+and MUST NOT occur elsewhere.  (That is, "*.example.com" and "*" are acceptable,
+but "www.*.com" and "w*.example.com" are not).
+
+    id-ce-allowedPrimaryDomain OBJECT IDENTIFIER ::=  { id-ce TBD1 }
+
+    AllowedPrimaryDomain ::= SEQUENCE SIZE (1..MAX) OF IA5String
+
+## Affiliated Domain Certificate Extension {#extension-affiliated}
+
+The Affiliated Domain extension allows groups of certificates under the same
+control to be identified without giving each certificate full identity with the
+domain.  The identities stated as affiliated domains are not bound to the
+subject of the certificate, but this extension indicates that there is a common
+management relationship between the subject and the affiliated domains.
+
+Because the affiliated domains are considered to be definitively bound to the
+public key, control of each affiliated domain MUST be verified by the CA using
+the same mechanisms as if the affiliated domain were listed in the
+subjectAltName extension.  Conforming CAs SHOULD mark the affiliatedDomain
+extension as non-critical.
+
+Each affiliated domain is represented as an IA5String. The name MUST be in the
+"preferred name syntax", as specified by Section 3.5 of {{!RFC1034}} and as
+modified by Section 2.1 of {{!RFC1123}}.  Note that while uppercase and lowercase
+letters are allowed in domain names, no significance is attached to the case. In
+addition, while the string " " is a legal domain name, affiliatedDomain
+extensions MUST NOT contain " ".  Rules for encoding internationalized domain
+names are specified in Section 7.2 of {{!RFC5280}}.
+
+Affiliated domains MAY be constrained in the same manner as subject
+distinguished names using the name constraints extension as described in Section
+4.2.1.10 of {{!RFC5280}}.
```

Comment:
If they are validated, then I believe that name constraints are necessary, not optional.

Reply:
Imported text from SAN, but I read this as saying that name constraints are permissible behavior (and the converse is that they MUST be supported).  I'll reword to clarify, unless you're arguing for name constraints being required to be present.

Reply:
If the CA is responsible for validating a name, then it can't validate a name outside of its name constraints.  For instance, if an enterprise were delegated *.enterprise.example, then it wouldn't be permitted to say that foo.example.com was OK.